Cargo.toml

```
@@ -20,12 +20,20 @@ exclude = ["ci"]
 
 [dependencies]
 rayon-core = { version = "1.12.0", path = "rayon-core" }
+wasm_sync = { version = "0.1.0", optional = true }
 
 # This is a public dependency!
 [dependencies.either]
 version = "1.0"
 default-features = false
 
+[features]
+# This feature switches to a spin-lock implementation on the browser's
+# main thread to avoid the forbidden `atomics.wait`.
+#
+# Only useful on the `wasm32-unknown-unknown` target.
+web_spin_lock = ["wasm_sync", "rayon-core/web_spin_lock"]
```

This should avoid the automatic feature for the optional dependency:
```suggestion
web_spin_lock = ["dep:wasm_sync", "rayon-core/web_spin_lock"]
```

(and FWIW you can do that with features of the same name too.)